Provenance-Emu/Provenance
# XCode 12 support

### What does this PR do
Makes sure that you can build with Xcode 12

This PR makes two changes, first we upgrade `realm-cocoa` to v5.5.0, which has works with XCode 12.
The other thing is adding `carthage.sh` from https://github.com/Carthage/Carthage/blob/master/Documentation/Xcode12Workaround.md to work around the current build issues with carthage.

So, with this PR you should be able to build by running `carthage.sh` instead of `carthage`, AKA running something like: `./carthage.sh bootstrap --platform tvOS --project-directory ./ --cache-builds` for tvOS builds.

This should solve #1437, although I think we should keep it open until this is fixed in upstream carthage, seems like they are working on it https://github.com/Carthage/Carthage/issues/3019

Fixes #1432
Fixes #1437
Fixes #1445 

### How should this be manually tested
I've launched it, but hasn't had time to try it out a lot, the realm update might break something, but I think that it should be fairly safe. But more testing would of course be nice
